python/ray/workflow/tests/test_error_handling.py

```diff
@@ -0,0 +1,136 @@
+import pytest
+
+import ray
+from ray import workflow
+
+
+def test_step_failure(workflow_start_regular_shared, tmp_path):
```

Comment:
Do we have test to make sure the task is not rerun by ray, like the data is loaded from storage and not reconstructed by the lineage?

Reply:
not really. it there an test example in Ray? I feel it is a bit tricky to trigger lineage reconstruction in a unittest -- this requires deleting an object in object store (so the task could be actually reconstructed). do you have any ideas (do we have related helper functions)?

Reply:
I think you can do this:

- a cluster with two nodes
- kill the node where the object ref is generated (thus the data is gone). write a flag to fs indicate it only run once.
- task does not rerun